### ISSUE TITLE
Black screen when open depth

### ISSUE THREAD

**Issue by biloba123** [google-ar/arcore-android-sdk#1183]
### SPECIFIC ISSUE ENCOUNTERED
When creating ARCore session that supports camera sharing and set depth mode to automatic if supported, I encountered camera preview black screen in ar mode.

### VERSIONS USED
- Android Studio: 4.1.1
- ARCore SDK for Android: 1.21.0
- Device manufacturer, model, and O/S: Google Pixel 4, android 11
- Google Play Services for AR (ARCore):  1.22.203220503
- Output of `adb shell getprop ro.build.fingerprint`: google/flame/flame:11/RQ1A.201205.008/6943376:user/release-keys

### STEPS TO REPRODUCE THE ISSUE
 1. Open shared_camera_java sample
 2. Set depth mode to Config.DepthMode.AUTOMATIC if supported
```
Config config = sharedSession.getConfig();
config.setFocusMode(Config.FocusMode.AUTO);
if (sharedSession.isDepthModeSupported(Config.DepthMode.AUTOMATIC)) {
  config.setDepthMode(Config.DepthMode.AUTOMATIC);
}
sharedSession.configure(config);
```
 3. Run app on device which supported depth api and switch to ar mode.

**Comment by devbridie** [comment]
I think this is the same issue as #1176 -- problem is known and on the way. Does the suggested workaround work for you, in this case?

**Comment by biloba123** [comment]
The workaround work for me, but there is another problem. The camera texture and ARCore's state are no longer synchronized. ARCore's frame's timestamp is behind for about 30～60ms.

**Comment by biloba123** [comment]
It looks like in Session#update, camera texture was updated and reattached to anchor gl context.  Confused, What is the role of the GPU stream inside ARCore?

**Comment by devbridie** [comment]
This issue has been fixed in 1.23. If you're still seeing this, please let me know.

**Comment by biloba123** [comment]
@devbridie This problem still exists in 1.23 and 1.24(EAP version) too.
